### PR TITLE
Be stricter about macro XML_CONTEXT_BYTES at build time

### PR DESCRIPTION
**Pull request by hartwork**

### CODE REVIEW

**Review by Snild-Sony — APPROVED** [review]
LGTM (but note that I have virtually no experience with cmake or autoconf).

**Comment by hartwork** [comment]
@Snild-Sony thanks for the review!

In the meantime I noticed that I need to do a tiny bit more to reject empty `#define XML_CONTEXT_BYTES`, somthing like this:

```c
#if defined(FOO) && (1 - FOO - 1 == 2) && (FOO - 0 != -2)
# warning FOO is defined and empty
#endif
```

Will adjust in a minute.

**Comment by Snild-Sony** [comment]
Do we need it to be that precise? Won't defining it to an empty value already fail compilation (or even preprocessing), e.g. by causing invalid syntax where it's used?

**Comment by hartwork** [comment]
@Snild-Sony that's one way to look at it but…

> error: operator '<' has no left operand

…seems more unfriendly than necessary. It's probably only a matter of time until we get bug reports for it.

Pushed.

File: expat/lib/xmlparse.c [hunk 1]
```diff
@@ -63,6 +63,10 @@
 
 #include "expat_config.h"
 
+#if ! defined(XML_CONTEXT_BYTES) || (1 - XML_CONTEXT_BYTES - 1 == 2) || (XML_CONTEXT_BYTES + 0 < 0)
```
Comment (Snild-Sony) [inline expat/lib/xmlparse.c:66]: Doesn't clang-format want to reformat this?

Comment (Snild-Sony) [reply expat/lib/xmlparse.c]: No, I was just surprised that clang-format didn't break the `#error` line. But if it doesn't, we're all good.

**Pull request merged**